apis/stacks/v1alpha1/types_test.go

```diff
@@ -28,35 +28,79 @@ var (
 	_ StackInstaller = &ClusterStackInstall{}
 )
 
-func TestStackInstallSpec_Image(t *testing.T) {
+func TestStackInstallSpec_ImageWithSource(t *testing.T) {
+	type want struct {
+		url string
+	}
+
 	tests := []struct {
-		name string
-		spec StackInstallSpec
-		want string
+		name   string
+		reason string
+		spec   StackInstallSpec
+		want   want
 	}{
 		{
-			name: "NoPackageSource",
+			name:   "NoPackageSource",
+			reason: "Packages without StackInstall source should be unchanged",
+			spec: StackInstallSpec{
+				Package: "cool/package:rad",
+			},
+			want: want{
+				url: "cool/package:rad",
+			},
+		},
+		{
+			name:   "PackageSourceSpecified",
+			reason: "Sourceless packages should inherit StackInstall source",
 			spec: StackInstallSpec{
+				Source:  "foo.bar",
 				Package: "cool/package:rad",
 			},
-			want: "cool/package:rad",
+			want: want{
+				url: "foo.bar/cool/package:rad",
+			},
 		},
 		{
-			name: "PackageSourceSpecified",
+			name:   "DockerSourceSpecified",
+			reason: "Docker hub sources should not have special treatment",
 			spec: StackInstallSpec{
 				Source:  "registry.hub.docker.com",
 				Package: "cool/package:rad",
 			},
-			want: "registry.hub.docker.com/cool/package:rad",
+			want: want{
+				url: "registry.hub.docker.com/cool/package:rad",
+			},
+		},
+		{
+			name:   "SourceWithProtocol",
+			reason: "Sources should honor host, port, and prefix",
+			spec: StackInstallSpec{
+				Source:  "insecure:3000/prefix/",
+				Package: "cool/tagless-package",
+			},
+			want: want{
+				url: "insecure:3000/prefix/cool/tagless-package",
+			},
+		},
+		{
```

Comment:
how about another test case where the package is already a fully qualified image with a source?  e.g.:
```go
name:   "PackageContainsSourceAlready",
reason: "Packages that contain a source already should have that source honored",
spec: StackInstallSpec{
	Source:  "foo.bar",
	Package: "my.registry/cool/tagless-package",
},
want: want{
	url: "my.registry/cool/tagless-package",
},
```